packages/headless/src/controllers/product-recommendations/headless-frequently-viewed-same-category.ts

```diff
@@ -0,0 +1,44 @@
+import {Schema, SchemaValues} from '@coveo/bueno';
+import {
+  baseProductRecommendationsOptionsSchema,
+  buildBaseProductRecommendationsList,
+} from './headless-base-product-recommendations';
+import {validateOptions} from '../../utils/validate-payload';
+import {ProductRecommendationEngine} from '../../app/product-recommendation-engine/product-recommendation-engine';
+
+const optionsSchema = new Schema({
+  ...baseProductRecommendationsOptionsSchema,
+});
+
+export type FrequentlyViewedSameCategoryListOptions = SchemaValues<
+  typeof optionsSchema
+>;
+
+export interface FrequentlyViewedSameCategoryListProps {
+  options?: FrequentlyViewedSameCategoryListOptions;
+}
+
+export type FrequentlyViewedSameCategoryList = ReturnType<
+  typeof buildFrequentlyViewedSameCategoryList
+>;
+export type FrequentlyViewedSameCategoryListState =
+  FrequentlyViewedSameCategoryList['state'];
+
+export const buildFrequentlyViewedSameCategoryList = (
```

Comment:
What is the difference between this controller and the one above? Aside from the name, it seems like the logic is identical?

Shouldn't there be some logic that makes the controllers behave differently, recommending within vs. in other categories?

Reply:
Indeed they're identical , the only difference is their `id`.

I guess I'm pretty naive but I thought all that matter was the `id` of the strategy on the Coveo platform.

Reply:
Ok, if the id is enough, then all might be good. Are the strategy string ids documented somewhere? Maybe [this table](https://docs.coveo.com/en/3284/coveo-for-commerce/product-recommendations-deployment) should be updated?

Reply:
I was not aware of this table. Thanks for sharing I'll update it !